Ms2Database/Controllers/EquipmentManager.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Ms2Database.DbClasses;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace Ms2Database.Controllers
+{
+    public class EquipmentManager
+    {
+        private readonly Ms2DbContext _context = new Ms2DbContext(); //Opens connection to database
+
+        public void EquipItem(long characterId, long uid, string slot, bool isCosmetic = true)
+        {
+            Item item = _context.Items.FirstOrDefault(column => column.UniqueId == uid); // Query for item
+            Item equippedItem;
+
+            if (isCosmetic) // Handles cosmetic
+            {
+                equippedItem = _context.Items.Include(table => table.Inventory)
+                                             .Where(column => column.Inventory.CharacterId == characterId)
+                                             .FirstOrDefault(column => column.CosmeticSlot == slot.ToUpper());
+
+                if (equippedItem == null || equippedItem.CosmeticSlot == "")
+                {
+                    item.CosmeticSlot = slot.ToUpper(); // Sets the item slot location
+                }
+                else // Unequips item occupying slot and sets incoming item slot location
+                {
+                    UnequipItem(equippedItem.UniqueId, true);
+                    item.CosmeticSlot = slot.ToUpper();
+                }
+            }
+            else // Handles regular equipment
+            {
+                equippedItem = _context.Items.Include(table => table.Inventory)
+                                             .Where(column => column.Inventory.CharacterId == characterId)
+                                             .FirstOrDefault(column => column.EquipmentSlot == slot.ToUpper());
+
+                if (equippedItem == null || equippedItem.EquipmentSlot == "")
+                {
+                    item.EquipmentSlot = slot.ToUpper();
+                }
+                else
+                {
+                    UnequipItem(equippedItem.UniqueId, true);
+                    item.EquipmentSlot = slot.ToUpper();
```

Comment:
Seems like you can move this and line 44 to after the if/else, then maybe invert line 42 to call line 48

Reply:
Gone ahead with the suggestion.